samtranslator/model/eventsources/pull.py

```diff
@@ -283,3 +291,147 @@ def get_policy_statements(self):
             }
             document["PolicyDocument"]["Statement"].append(kms_policy)
         return [document]
+
+
+class SelfManagedKafka(PullEventSource):
+    """
+    SelfManagedKafka event source
+    """
+
+    resource_type = "SelfManagedKafka"
+    AUTH_MECHANISM = ["SASL_SCRAM_256_AUTH", "SASL_SCRAM_512_AUTH", "BASIC_AUTH"]
+
+    def get_policy_arn(self):
+        return None
+
+    def get_policy_statements(self):
+        if not self.KafkaBootstrapServers:
+            raise InvalidEventException(
+                self.relative_id,
+                "No KafkaBootstrapServers provided for self managed kafka as an event source",
+            )
+
+        if not self.Topics:
+            raise InvalidEventException(
+                self.relative_id,
+                "No Topics provided for self managed kafka as an event source",
+            )
+
+        if len(self.Topics) != 1:
+            raise InvalidEventException(
+                self.relative_id,
+                "Topics for self managed kafka only supports single configuration entry.",
+            )
+
+        if not self.SourceAccessConfigurations:
+            raise InvalidEventException(
+                self.relative_id,
+                "No SourceAccessConfigurations for self managed kafka event provided.",
+            )
+        document = self.generate_policy_document()
+        return [document]
+
+    def generate_policy_document(self):
+        document = {
+            "PolicyDocument": {
+                "Statement": [],
+                "Version": "2012-10-17",
+            },
+            "PolicyName": "SelfManagedKafkaExecutionRolePolicy",
+        }
+
+        authentication_uri, has_vpc_config = self.get_secret_key()
+        if authentication_uri:
+            secret_manager = self.get_secret_manager_secret(authentication_uri)
+            document["PolicyDocument"]["Statement"].append(secret_manager)
+
+        if has_vpc_config:
+            vpc_permissions = self.get_vpc_permission()
+            document["PolicyDocument"]["Statement"].append(vpc_permissions)
+
+        if self.SecretsManagerKmsKeyId:
+            kms_policy = self.get_kms_policy()
+            document["PolicyDocument"]["Statement"].append(kms_policy)
+
+        return document
+
+    def get_secret_key(self):
+        authentication_uri = None
+        has_vpc_subnet = False
+        has_vpc_security_group = False
+        for config in self.SourceAccessConfigurations:
+            if config["Type"] == "VPC_SUBNET":
```

Comment:
Can we use `config.get("Type")` to avoid throwing 500 errors on invalid templates?
```
auth_type = config.get("Type")
if auth_type == "VPC_SUBNET":
...
```